### PR TITLE
How to work on jira tickets

### PR DESCRIPTION
**Pull request by alastair**
Some rough text on how to work on jira tickets. Mostly intended to try and stop the "can I work on this?" comments that we're getting.

Does it make sense to call this Jira.md or does it fall into the wider Bugs.md

Do we also want "how to report bugs", or is that more suited to the wiki?

I wrote the workflow text based on the workflow that we have in AB. I don't know if this is the same in other projects. I seem to recall that they all use the same workflow, so I hope it's ok. If not, please propose prose that better suits what we do/try to do.

Do we want so much text, would a pretty diagram also be useful?

### CODE REVIEW

**Comment by Freso** [comment]
> Do we also want "how to report bugs", or is that more suited to the wiki?

FWIW, this is already on the wiki: https://wiki.musicbrainz.org/How_to_Report_an_Issue

**Review by Freso — APPROVED** [review]
LGTM, FWIW.

File: Jira.md
Comment (mwiencek) [inline Jira.md]: Maybe it'd be good to mention that if something is marked "Decision Required," they should resolve any open questions first. I think this status is specific to our workflow.

File: Jira.md
Comment (mwiencek) [inline Jira.md]: I noticed there are three different capitalizations of "pull request" here. :)

```
Pull request
Pull Request
pull request
```

File: Jira.md
Comment (mwiencek) [inline Jira.md]: In MBS at least, they have to click "In Code Review." I guess it's different for other projects.

**Comment by gentlecat** [comment]
> > Do we also want "how to report bugs", or is that more suited to the wiki?
>
> FWIW, this is already on the wiki: https://wiki.musicbrainz.org/How_to_Report_an_Issue

Perhaps this page can be generalised to apply to all projects and maybe even moved there?

> Does it make sense to call this Jira.md or does it fall into the wider Bugs.md

Or maybe even *Ticket_tracking.md*?

File: Jira.md
Comment (gentlecat) [inline Jira.md]: .

File: Jira.md
Comment (gentlecat) [inline Jira.md]: Maybe also link to https://tickets.metabrainz.org/secure/BrowseProjects.jspa.

File: Jira.md
Comment (gentlecat) [inline Jira.md]: Should this be `## Working on tickets`?

File: Jira.md
Comment (gentlecat) [reply Jira.md]: Some projects have the same workflow set on them. Perhaps we should make it a default.

File: Jira.md
Comment (gentlecat) [inline Jira.md]: It would probably more correct to say it like this:

*Once a pull request has been reviewed and merged by one of the project maintainers, it can be closed and marked as "Fixed".*

File: Jira.md
Comment (gentlecat) [reply Jira.md]: Also, I assume *ticket* and *bug* refer to the same thing. We should settle on one.

File: Jira.md
Comment (gentlecat) [inline Jira.md]: > Project leads will automatically get sent an email informing them of your comment

Not necessarily. People might not receive an email depending on how their notifications are configured. For example, I don't receive emails about every CB ticket. I do receive email if something happens to tickets that I'm watching or ones that I opened.

File: Jira.md
Comment (gentlecat) [inline Jira.md]: `"In progress"`?

File: Jira.md
Comment (alastair) [reply Jira.md]: I like ticket, because it's broader (task, improvement, etc)

File: Jira.md
Comment (alastair) [reply Jira.md]: I think the workflow for AB and LB is the same, but I'm not sure about the others. Obviously MBS is different. Can someone provide us with a list of what project has which, and what the steps are? @Freso? I don't know if this is something that can only be seen by an admin or by everyone.

Comment (gentlecat) [reply Jira.md]: screenshot 2017-03-13 21 57 35
screenshot 2017-03-13 21 58 51

File: Jira.md
Comment (gentlecat) [reply Jira.md]: I agree.

File: Jira.md
Comment (alastair) [reply Jira.md]: My reason for this is that I want to stop people from telling me on IRC "hey I did something, can you look". If this isn't global we should do something else. Perhaps just saying this is good enough?

Comment (gentlecat) [reply Jira.md]: Saying something like "People who are watching this ticket will automatically..." will be more correct I think.

File: Jira.md
Comment (alastair) [reply Jira.md]: well. at least there's *only* 2... Do we need to bring this up at a meeting? I note that there are only 2 webapp projects on default. i wonder if they're just on default because no one changed them.

Can you post an image or text description of the MusicBrainz workflow? (from the MBS equivalent of https://tickets.metabrainz.org/plugins/servlet/project-config/AB/workflows).

Comment (gentlecat) [reply Jira.md]: screenshot 2017-03-13 22 39 28

**Comment by alastair** [comment]
Covered all points except the workflow stuff.
I'm happy to move AB and LB to the musicbrainz workflow, which means that all main webapps would use the same workflow and we can refer to it explicitly in this doc.

**Comment by gentlecat** [comment]
Do we keep it `Jira.md`?

**Review by gentlecat — APPROVED** [review]

**Comment by alastair** [comment]
I thought about that - if we consider that contributing guidelines are for people who want to help develop our apps then jira is the tool that they need to interact with for this task. I don't mind keeping it jira.

**Comment by gentlecat** [comment]
Alright.

File: Jira.md
Comment (eebigger) [reply Jira.md]: I donno if this helps/is needed--but I made a diagram to discuss my workflow--I could easily adapt it. 
workflow

**Pull request merged**
